pkg/builder/openapi.go

```diff
@@ -152,7 +152,17 @@ func (o *openAPI) finalizeSwagger() (*spec.Swagger, error) {
 		}
 	}
 
-	return o.swagger, nil
+	sp := o.swagger
+	names, parameters, err := collectSharedParameters(sp)
+	if err != nil {
+		return nil, err
+	}
+
+	if sp.Parameters != nil {
+		return nil, fmt.Errorf("shared parameters already exist") // should not happen with the builder, but to be sure
```

Comment:
Should we panic if this isn't supposed to happen?

Reply:
good question. Is panic what we usually do in Kube? No strong opinion, but I think I prefer an error.